practice/src/scripts/models/task.model.ts

```diff
@@ -70,6 +82,103 @@ class TaskModel {
             storage.saveTasks(this.tasks);
         }
     };
+
+
+    toggleTask(id: number, type: string, renderTasks: (tasks: Task[]) => void) {
+        const currentTime = new Date();
+        const hours = currentTime.getHours();
+        const formattedHours = hours % 24 || 24;
+        const minutes = currentTime.getMinutes();
+        const seconds = currentTime.getSeconds();
+        const formattedTime = `${formattedHours}:${minutes}:${seconds.toString().padStart(2, '0')}, ${currentTime.toLocaleDateString('en-US', {
+            year: 'numeric',
+            month: 'short',
+            day: 'numeric',
+        })}`;
+    
+        switch (type) {
+            case 'active':
+                this.tasks.forEach((task: Task) => {
+                    if (task.id === id) {
+                        task.isCompleted = true;
+                        task.updatedAt = formattedTime;
+                    }
+                });
+                break;
+            case 'unactive':
+                this.tasks.forEach((task: Task) => {
+                    if (task.id === id) {
+                        task.isCompleted = false;
+                        task.updatedAt = formattedTime;
+                    }
+                });
+                break;
+            case 'toggle':
+                const checkTask = this.tasks.find((item: Task) => !item.isCompleted);
+                this.tasks.forEach((task: Task) => {
+                    if (checkTask) {
+                        if (!task.isCompleted) {
+                            task.isCompleted = true;
+                        }
+                    } else {
+                        task.isCompleted = false;
+                    }
+                    task.updatedAt = formattedTime;
+                });
+                break;
+            default:
+                break;
+        }
+    
+        const locationHash = window.location.hash;
+        locationHash === '#completed'
+            ? this.filterTask('completed', renderTasks)
+            : locationHash === '#active'
+            ? this.filterTask('active', renderTasks)
+            : renderTasks(this.tasks);
+    
+        storage.saveTasks(this.tasks);
+    }
+    
+    checkAllToggleTask = (renderTasks: (tasks: Task[]) => void): void => {
+        this.toggleTask(0, "toggle", renderTasks);
```

Comment:
use object params avoid wrong params passing
this.toggleTask({ 0, "toggle", renderTasks });